std/range/package.d

```diff
@@ -1475,6 +1481,15 @@ private struct ChooseResult(R1, R2)
         }
     }
 
+    void opAssign(return scope ChooseResult r) @trusted
+    {
+        r1Chosen = r.r1Chosen;
+        if (r1Chosen)
+            r1 = r.r1;  // assigning to union members makes it @trusted
```

Comment:
Your implication in the comment is wrong. assigning to union members makes it _`@system`_, hence it need to be `@trusted`